lib/extract-js.js

```diff
@@ -0,0 +1,216 @@
+var htmlparser = require('htmlparser2');
+var Errors = require('./errors');
+
+/**
+ * Html file representation (needs for errors output).
+ *
+ * @name HtmlFile
+ */
+var HtmlFile = function(filename, source) {
+    this._filename = filename;
+    this._lines = source.split(/\r\n|\r|\n/);
```

Comment:
Can this regex be substituted with `/\r\n?|\n\r?/`? If so, extract this regex to a closed over var near the top of the file. It's used in 3 places.